folding-schemes/src/folding/hypernova/circuit.rs

```diff
@@ -1,286 +1,134 @@
+            current_gamma *= gamma.clone();
         }
-        Ok(result)
     }
 
-    /// Computes the sum that the verifier has to compute at section 5, step 5 of "A multi-folding scheme for CCS".
-    ///
-    /// # Arguments
-    /// - `cs`: constraint system
-    /// - `ccs`: the CCS instance
-    /// - `vec_sigmas`: vector of $\sigma_j$ values
-    /// - `vec_thetas`: vector of $\theta_j$ values
-    /// - `gamma`: value $\gamma$
-    /// - `beta`: vector of $\beta_j$ values
-    /// - `vec_r_x`: vector of $r_{x_j}$ values
-    /// - `vec_r_x_prime`: vector of $r_{x_j}^{\prime}$ values
-    ///
-    /// # Notes
-    /// Arguments to this function are *almost* the same as the arguments to `compute_c_from_sigmas_and_thetas` in `utils.rs`.
-    #[allow(clippy::too_many_arguments)]
-    pub fn compute_c_from_sigmas_and_thetas(
-        cs: ConstraintSystemRef<C::ScalarField>,
-        ccs: &CCS<C>,
-        vec_sigmas: Vec<Vec<FpVar<C::ScalarField>>>,
-        vec_thetas: Vec<Vec<FpVar<C::ScalarField>>>,
-        gamma: FpVar<C::ScalarField>,
-        beta: Vec<FpVar<C::ScalarField>>,
-        vec_r_x: Vec<Vec<FpVar<C::ScalarField>>>,
-        vec_r_x_prime: Vec<FpVar<C::ScalarField>>,
-    ) -> Result<FpVar<C::ScalarField>, SynthesisError> {
-        let mut c =
-            FpVar::<C::ScalarField>::new_witness(cs.clone(), || Ok(C::ScalarField::zero()))?;
-        let t = FpVar::<C::ScalarField>::new_witness(cs.clone(), || {
-            Ok(C::ScalarField::from(ccs.t as u64))
-        })?;
-
-        let mut e_lcccs = Vec::new();
-        for r_x in vec_r_x.iter() {
-            let e_1 = EqEvalGadget::eq_eval(r_x.to_vec(), vec_r_x_prime.to_vec())?;
-            e_lcccs.push(e_1);
-        }
-
-        for (i, sigmas) in vec_sigmas.iter().enumerate() {
-            let i_var = FpVar::<C::ScalarField>::new_witness(cs.clone(), || {
-                Ok(C::ScalarField::from(i as u64))
-            })?;
-            let pow = i_var * t.clone();
-            c += Self::sum_muls_gamma_pows_eq_sigma(
-                gamma.clone(),
-                e_lcccs[i].clone(),
-                sigmas.to_vec(),
-                pow,
-            )?;
-        }
-
-        let mu = FpVar::<C::ScalarField>::new_witness(cs.clone(), || {
-            Ok(C::ScalarField::from(vec_sigmas.len() as u64))
-        })?;
-        let e_2 = EqEvalGadget::eq_eval(beta, vec_r_x_prime)?;
-        for (k, thetas) in vec_thetas.iter().enumerate() {
-            // get prepared thetas. only step different from original `compute_c_from_sigmas_and_thetas`
-            let mut prepared_thetas = Vec::new();
-            for i in 0..ccs.q {
-                let prepared: Vec<FpVar<C::ScalarField>> =
-                    ccs.S[i].iter().map(|j| thetas[*j].clone()).collect();
-                prepared_thetas.push(prepared.to_vec());
+    let ccs_c = Vec::<FpVar<F>>::new_constant(cs.clone(), ccs.c.clone())?;
+    let e_k = EqEvalGadget::eq_eval(beta, vec_r_x_prime)?;
```

Comment:
The $e_2$ is what the paper uses to describe the $e$ used for the CCCS instance, because the paper only describes the approach for 1 LCCCS instance and 1 CCCS instance.
In our case we work with $\mu$ LCCCS instances and $\nu$ CCCS instances, so the $e_2$ could fall into the LCCCS instances, where here is a CCCS instance, thus I used $e_k$ just as identifier to make it agnostic to a specific subindex.
(I've seen now $e_1$ few lines above this one and updated it to use also an agnostic subindex.)